src/imp.rs

```diff
@@ -644,33 +728,49 @@ fn sign_stct<Rng: RngCore + CryptoRng>(
 fn stake_sign(
     sk: &SecretKey,
     pk: &PublicKey,
+    counter: u64,
     value: u64,
-    created_at: u64,
 ) -> Signature {
-    let mut bytes = Vec::with_capacity(16);
+    let mut msg = Vec::with_capacity(u64::SIZE + u64::SIZE);
+
+    msg.extend(counter.to_bytes());
+    msg.extend(value.to_bytes());
+
+    sk.sign(pk, &msg)
+}
+
+/// Creates a signature compatible with what the stake contract expects for a
+/// unstake transaction.
+fn unstake_sign(
+    sk: &SecretKey,
+    pk: &PublicKey,
+    counter: u64,
+    note: Note,
+) -> Signature {
+    let mut msg = Vec::with_capacity(u64::SIZE + Note::SIZE);
 
-    bytes.extend(value.to_bytes());
-    bytes.extend(created_at.to_le_bytes());
+    msg.extend(counter.to_bytes());
+    msg.extend(note.to_bytes());
 
-    sk.sign(pk, &bytes)
+    sk.sign(pk, &msg)
 }
 
 /// Creates a signature compatible with what the stake contract expects for a
 /// withdraw transaction.
 fn withdraw_sign(
     sk: &SecretKey,
     pk: &PublicKey,
-    stake: &StakeInfo,
-    note: &Note,
+    counter: u64,
+    address: StealthAddress,
+    nonce: BlsScalar,
 ) -> Signature {
-    let mut bytes = Vec::with_capacity(24 + Note::SIZE);
+    let mut msg = Vec::with_capacity(u64::SIZE + Note::SIZE);
```

Comment:
You're using the `Note::SIZE` but looks like you've removed the `note`.

Reply:
Good catch. Should really be `StealthAddress::SIZE + BlsScalar::SIZE`.